addons/mail/models/mail_thread.py

```diff
@@ -1652,11 +1652,11 @@ def _find_partner_from_emails(self, emails, res_model=None, res_id=None, check_f
             if not email_address:
                 partner_ids.append(partner_id)
                 continue
-            if exclude_aliases and self.env['mail.alias'].search([('alias_name', 'ilike', email_address)], limit=1):
+            email_address = email_address[0]
+            if exclude_aliases and self.env['mail.alias'].search([('alias_name', 'ilike', email_address.split('@')[0])], limit=1):
```

Comment:
This is not technically correct as you have to check the right part also (sales@another.domain is not the same as sales@odoo.com from alias point of view).

Something like
```
catchall_domain = ... # get catchall domain
email_left_part = email_address.split('@')[0] if email_address.split('@')[1].lower() == catchall_domain.lower()
```